### PR TITLE
backupccl: upgrade restoreTPCCIncLatest/nodes=10 backup fixture

### PR DESCRIPTION
**Pull request by msbutler**
restoreTPCCIncLatest/nodes=10 has been consistently failing because the backup fixture is more than one major version behind the cluster running the restore. This patch updates the backup fixture used by this roachtest.

Informs #88090 #89380

Release note: None

### CODE REVIEW

**Comment by msbutler** [comment]
passes in 17 mins!

File: pkg/cmd/roachtest/tests/restore.go [hunk 1]
```diff
@@ -471,7 +471,6 @@ func registerRestore(r registry.Registry) {
 		{dataSet: dataBank2TB{}, nodes: 10, timeout: 6 * time.Hour},
 		{dataSet: dataBank2TB{}, nodes: 32, timeout: 3 * time.Hour},
 		{dataSet: dataBank2TB{}, nodes: 6, timeout: 4 * time.Hour, cpus: 8, largeVolumes: true},
-		{dataSet: tpccIncData{}, nodes: 10, timeout: 6 * time.Hour},
```
Comment (adityamaru) [inline pkg/cmd/roachtest/tests/restore.go:474]: if we're not running tpccIncData do you think you could:

1. Remove tpccIncData
2. Rename tpccIncDataLatest back to tpccIncData

I had only added latest cause we didn't want to delete tpccIncData as there were some ongoing invesitgations.

**Review by adityamaru — APPROVED** [review]

File: pkg/cmd/roachtest/tests/restore.go (hunk 1 above)
Comment (adityamaru) [reply pkg/cmd/roachtest/tests/restore.go:474]: I think this would also mean that it'd reflect correctly in roachperf.

Comment (msbutler) [reply pkg/cmd/roachtest/tests/restore.go:474]: aha. Good point. done.

**Comment by msbutler** [comment]
TFTR!! 

bors r=adityamaru

**Comment by msbutler** [comment]
bors r=adityamaru

**Pull request merged**